rasa/core/test.py

```diff
@@ -434,7 +434,8 @@ def _create_data_generator(
     from rasa.shared.core.generator import TrainingDataGenerator
 
     tmp_domain_path = Path(tempfile.mkdtemp()) / "domain.yaml"
-    agent.domain.persist(tmp_domain_path)
+    domain = agent.domain if isinstance(agent.domain, Domain) else Domain.empty()
```

Comment:
Actually it needs to be `isinstance(agent.domain, Domain)` because it seems the type of domain parameter in the constructor of `Agent` is `Optional[Union[Text, Domain]]`, so checking `is not None` fails with `Item "str" of "Union[str, Domain]" has no attribute "persist" `

Reply:
something seems wrong there, because for instance, the `_create_tracker_store()` method called in `Agent.__init__` expects a `Domain` instance

Reply:
Yeah, I've looked up also all usages of `Agent.load` and nowhere did it require `domain` parameter as string. I'll remove this type hint.